src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -1414,7 +1414,7 @@ std::string GeneratedClassName(const Descriptor* desc) {
   std::string classname = desc->name();
   const Descriptor* containing = desc->containing_type();
   while (containing != NULL) {
-    classname = containing->name() + '_' + classname;
+    classname = containing->name() + '\\' + ClassNamePrefix(classname, desc) + classname;
```

Comment:
Could you add a test in which containing message has reserved name? For example:
```
message Empty {
  message Nested {}
}
```